### PR TITLE
Fixing the KMS key id confusion

### PR DESCRIPTION
**Pull request by aliaksei-ivanou**
*Issue #, if available:*

*Description of changes:*
There is a bug blocking a creation of a new KMS key with an arbitrary alias.

```
.resourceProvider(
       "arbitrary-kms-key",
        new blueprints.KmsKeyProvider("arbitrary-kms-key")
      )
```

Also, splitting the KMS key provider into two: LookupKeyProvider and CreateKeyProvioder

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by shapirov103 — APPROVED** [review]
lgtm

File: lib/stacks/eks-blueprint-stack.ts [hunk 1]
```diff
@@ -1,99 +1,100 @@
...
-import { IVpc } from 'aws-cdk-lib/aws-ec2';
-import { KubernetesVersion } from 'aws-cdk-lib/aws-eks';
-import { Construct } from 'constructs';
-import { MngClusterProvider } from '../cluster-providers/mng-cluster-provider';
-import { VpcProvider } from '../resource-providers/vpc';
-import * as spi from '../spi';
-import * as constraints from '../utils/constraints-utils';
-import * as utils from '../utils';
-import { cloneDeep } from '../utils';
+import * as cdk from "aws-cdk-lib";
```
Comment (shapirov103) [inline lib/stacks/eks-blueprint-stack.ts:1]: there is no way to tell what changed here.

**Review by shapirov103 — COMMENTED** [review]
@aliaksei-ivanou overall looks great, however I was not able to review the eks-blueprint-stack.ts because of the formatting changes. 

Let's do this: please set tabsize to 4 in your settings. Also configure your IDE to apply formatting only to the changed lines, otherwise review is very tedious. 

The fix, however, for the KMS Key provider is great. I need to remember that it will be a breaking change for some customers, so should be in the release notes.

**Comment by aliaksei-ivanou** [comment]
Changes are merged into https://github.com/aws-quickstart/cdk-eks-blueprints/pull/614

**Pull request closed without merging**